sbt-version-policy/src/main/scala/sbtversionpolicy/VersionChange.scala

```diff
@@ -0,0 +1,83 @@
+package sbtversionpolicy
+
+import coursier.version.{ Version, VersionCompatibility }
+
+sealed trait VersionChange
+object VersionChange {
+
+  /** Major upgrade represents change in a "major" version,
+   * for example `x` in `x.y.z` in SemVer where `x > 0`.
+   *
+   * There is NO source compatibility or binary compatibility
+   * guaratee between a major upgrade.
+   */
+  case object MajorUpgrade extends VersionChange
+
+  /** Minor upgrade represents change in a "minor" version,
+   * for example `y` in `x.y.z` in SemVer where `x > 0`.
+   *
+   * Binary compatibility MUST be guranteed between a minor upgrade.
+   */
+  case object MinorUpgrade extends VersionChange
+
+
+  /** A patch represents change in a "patch" version,
+   * for example `z` in `x.y.z` in SemVer where `x > 0`.
+   *
+   * Binary compatibility MUST be guranteed between a patch.
+   * Source compatibility [[https://docs.scala-lang.org/overviews/contributors/index.html SHOULD]]
+   * be guaranteed between a patch.
+   */
+  case object Patch extends VersionChange
+
+  /** Prerelease represents version changes during development period.
+   * This could include 0.y.z in SemVer or changes between release candidates.
+   *
+   * There is NO source compatibility or binary compatibility
+   * guaratee between a prerelease change.
+   */
+  case object Prerelease extends VersionChange
+
+  /** Same versions.
+   */
+  case object Same extends VersionChange
+
+  def apply(value1: String, value2: String, scheme: VersionCompatibility): VersionChange = {
```

Comment:
Also, I find it confusing that `VersionChange` contains concepts related to semantic versioning (e.g., `Patch`, `MinorUpgrade`, `MajorUpgrade`), but it can take any `VersionCompatibility` as a parameter (not just `EarlySemVer`).

Maybe what this type `VersionChange` is really about is the compatibility guarantees between version changes? So, I would rather have:

~~~ scala
sealed trait VersionChange
case object BinaryCompatible extends VersionChange
case object BinaryAndSourceCompatible extends VersionChange
case object Breaking extends VersionChange
~~~

Reply:
> What do we return between `1.0.0` and `1.0.0+3-123abc`?

Isn't it suppose to be `Same`?

Reply:
> Also, I find it confusing that VersionChange contains concepts related to semantic versioning (e.g., Patch, MinorUpgrade, MajorUpgrade), but it can take any VersionCompatibility as a parameter (not just EarlySemVer).

The way I see is that all the version schemes that we're dealing with have the same notion of changes (major change, minor change etc) or as Semantic Versioning or subset but just using different digits. I'm happy to use your `VersionChange` notion instead.

Reply:
Addressed in https://github.com/scalacenter/sbt-version-policy/pull/31/commits/db10da5f6647fdd12a61bdd0d3ab41133102e5cb